viewing_party/party.py

```diff
@@ -1,23 +1,160 @@
+    if not user_data["watched"] :
+        return None
+
+    genre_list = []
+    for movies in user_data["watched"]:
+        genre_list.append(movies["genre"])
 
+    dict = {}
+    for item in genre_list:
+        if item in dict:
+            dict[item] += 1
+        else:
+            dict[item] = 1
+    
+    max_value = 0
+    max_key = None
+
+    for key, value in dict.items():
+        if value > max_value:
+            max_value = value
+            max_key = key
+    return max_key
+  
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def user_movies_list(user_data):
+    user_movies = []
+    for movie in user_data["watched"]:
+        if movie not in user_movies:
+            user_movies.append(movie)
+    return user_movies
+
+def friend_movies_list(user_data):
+    friend_movies = []
+    for items in user_data["friends"]:
+        for movie in items["watched"]:
+            if movie not in friend_movies:
+                friend_movies.append(movie)
+    return friend_movies
+
+def get_unique_watched(user_data):
+    
+    user_movies_collection = user_movies_list(user_data)
+    friend_movies_collection = friend_movies_list(user_data)
+   
+    unique_movies = []
+    for movie in user_movies_collection:
+        if movie not in friend_movies_collection:
+            unique_movies.append(movie)
+    return unique_movies
+
+def get_friends_unique_watched(user_data):
+
+    user_movies_collection = user_movies_list(user_data)
+    friend_movies_collection = friend_movies_list(user_data)
+
+    friends_unique_movies = []
+
+    for movies in friend_movies_collection:
+        if movies not in user_movies_collection:
+            friends_unique_movies.append(movies)
+
+    return friends_unique_movies
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+
+    recomanded_movies = []
+    friend_movies_collection = friend_movies_list(user_data)
+
+    for movie in friend_movies_collection:
+        if movie["host"] in user_data["subscriptions"] and movie not in user_data["watched"] and movie not in recomanded_movies:
+            recomanded_movies.append(movie)
+    return recomanded_movies         
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+
+    recomanded_movie = []
+    if  not user_data["watched"]  or not user_data["friends"]:
+        return recomanded_movie 
+
+    most_genre = get_most_watched_genre(user_data)
+    
+    friend_movies_collection = friend_movies_list(user_data)
+    
+    for movie in friend_movies_collection:
+        if movie not in user_data["watched"] and movie["genre"]  in most_genre and movie not in recomanded_movie :

```

Comment:
How could we reuse our earlier function `get_friends_unique_watched` to simplify our code here?